File: server/src/routes/ideAgent.js
import express from 'express';
import { CodeAgent } from '../agents/codeAgent.js';
import { ReviewAgent } from '../agents/reviewAgent.js';
import { PlannerAgent } from '../agents/plannerAgent.js';

const router = express.Router();
const codeAgent = new CodeAgent();
const reviewAgent = new ReviewAgent();
const plannerAgent = new PlannerAgent();

// Thin endpoint for the IDE Agent Panel — reuses the existing agent classes
// as-is (no new agent logic), just exposes each one standalone instead of
// only through the chat pipeline's classifyTask routing.
router.post('/', async (req, res) => {
  const { agent, input } = req.body;
  if (!agent || !input) {
    return res.status(400).json({ error: 'agent and input are required' });
  }

  try {
    if (agent === 'code') {
      const result = await codeAgent.run(input);
      return res.json({ output: result.output.answer, confidence: result.output.confidence });
    }

    if (agent === 'planner') {
      const result = await plannerAgent.run(input);
      const plan = result.output;
      const md = `# Intent\n${plan.intent}\n\n# Task Type\n${plan.taskType}\n\n# Steps\n${plan.steps.map((s, i) => `${i + 1}. ${s}`).join('\n')}`;
      return res.json({ output: md });
    }

    if (agent === 'review') {
      const critique = await reviewAgent.critique(input, {
        question: input,
        evidenceSummary: input,
        contradictions: []
      });
      const md = `# Verdict\n${critique.pass ? 'Pass' : 'Issues found'}\n\n# Issues\n${critique.issues.length ? critique.issues.map(i => `- ${i}`).join('\n') : '- None'}\n\n# Confidence\n${critique.confidenceScore}% — ${critique.confidenceReason}`;
      return res.json({ output: md });
    }

    return res.status(400).json({ error: `Unknown agent: ${agent}` });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

export default router;

File: server/app/routes/ide_agent.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

from ..agents.code_agent import CodeAgent
from ..agents.planner_agent import PlannerAgent
from ..agents.review_agent import ReviewAgent

router = APIRouter()
code_agent = CodeAgent()
review_agent = ReviewAgent()
planner_agent = PlannerAgent()


class IdeAgentBody(BaseModel):
    agent: str | None = None
    input: str | None = None


@router.post("/api/ide/agent")
async def ide_agent(body: IdeAgentBody):
    if not body.agent or not body.input:
        raise HTTPException(status_code=400, detail="agent and input are required")

    try:
        if body.agent == "code":
            result = await code_agent.run(body.input)
            return {"output": result["output"]["answer"], "confidence": result["output"]["confidence"]}

        if body.agent == "planner":
            result = await planner_agent.run(body.input)
            plan = result["output"]
            steps = "\n".join(f"{i + 1}. {s}" for i, s in enumerate(plan["steps"]))
            md = f"# Intent\n{plan['intent']}\n\n# Task Type\n{plan['taskType']}\n\n# Steps\n{steps}"
            return {"output": md}

        if body.agent == "review":
            critique = await review_agent.critique(body.input, question=body.input, evidence_summary=body.input, contradictions=[])
            issues = "\n".join(f"- {i}" for i in critique["issues"]) if critique["issues"] else "- None"
            md = f"# Verdict\n{'Pass' if critique['pass'] else 'Issues found'}\n\n# Issues\n{issues}\n\n# Confidence\n{critique['confidenceScore']}% — {critique['confidenceReason']}"
            return {"output": md}

        raise HTTPException(status_code=400, detail=f"Unknown agent: {body.agent}")
    except HTTPException:
        raise
    except Exception as err:
        raise HTTPException(status_code=500, detail=str(err))
